geoh5py/objects/object_base.py

```diff
@@ -415,6 +459,15 @@ def property_groups(self) -> list[PropertyGroup] | None:
         """
         List of :obj:`~geoh5py.groups.property_group.PropertyGroup`.
         """
+        property_groups = [
+            child for child in self.children if isinstance(child, PropertyGroup)
+        ]
+
+        if self._property_groups is not None:
+            self._property_groups = list(set(self._property_groups + property_groups))
+        elif len(property_groups) > 0:
+            self._property_groups = property_groups
+
```

Comment:
Why do we need to check if `self._property_group` is not None and append?
Can maybe simplify this with:

```suggestion
        if self._property_groups is None:
            property_groups = [
                child for child in self.children if isinstance(child, PropertyGroup)
            ]

            if len(property_groups) > 0:
                self._property_groups = property_groups
                
        return self._property_groups

```

Reply:
This function handle the scenario where a property group is in children but not in the _property_group. I don't know when it can occurs, but the function was made to handle this scenario.

If this scenario is impossible, we could choose to not handle it, will make the function simpler.

Reply:
might need to double check that it behaves as intended.